db/python/tables/analysis.py

```diff
@@ -295,20 +295,19 @@ async def query(self, filter_: AnalysisFilter) -> List[AnalysisInternal]:
         else:
             _query = f"""
             SELECT a.id as id, a.type as type, a.status as status,
```

Comment:
Can we potentially unify these two branches with JSON_ARRAYAGG, bevause we're currently not returning cohorts here (if you don't include a cohort in the filter), and there's a separate call to get sequencing-group-ids. 

IMO we should always LEFT JOIN on cohorts AND sequencing-groups, and then always JSON_ARRAYAGG so we're correctly returning the right metadata. We should probably also change the methods below to also return cohorts, otherwise I bet we're going to have an issue with "this endpoint is returning no cohorts when it should have them".

Reply:
It actually does return cohorts, just a bit further down in a separate query:
https://github.com/populationgenomics/metamist/pull/791/files#diff-64b0b1278d5707365163edea17225c7b615255c0f4262d1e477b8456fc0f40bcL314

You're definitely right that it would be cleaner to be combined into one query though, I'll make that update

Reply:
The two branches was an unfortunate hack: if cohorts aren't part of the query, then joining against the cohort table would seem to ensure zero rows returned. So we made two branches as an expedient solution. If we had spent longer on it we would have preferred implementing this as a single query, and look forward to seeing this update.

Reply:
Having a `LEFT JOIN` should mean that analyses are returned even if there are no cohorts or sequencing groups in the join tables